cmake/FindCereal.cmake

```diff
@@ -6,6 +6,7 @@ endif()
 
 if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/third_party/cereal/include")
   add_library(cereal INTERFACE)
-
   target_include_directories(cereal SYSTEM INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/third_party/cereal/include")
+  target_compile_definitions(cereal INTERFACE CEREAL_THREAD_SAFE=1)
```

Comment:
this forces all the libraries that use it to use the thread safe mode, which is better